### ISSUE TITLE
Can't download BBC content in GiA on macOS, works in Parallels Windows 10 VM

### ISSUE THREAD

**Issue by dubbayoo** [Ascoware/get-iplayer-automator#200]
Just wondering if there's some weird thing I'm doing wrong, but GiA (and for that matter, get_iplayer via CLUI) on macOS 10.12.6 won't download BBC content, but using get_iplayer on the same computer inside a Windows 10 Parallels (v14, also had the issue with v13) VM works quite fine. Output will be pasted at the end of the message.

- MacBookPro13,3 (15-inch, Late 2016)
- macOS 10.12.6
- NordVPN in use with an active UK exit point (which number doesn't seem to matter)
- CAN view iPlayer content in Safari without issue
- Windows 10 Parallels VM works to download content, macOS GiA wrapper fails with "program may no longer be available" message
- Windows VM shares the macOS network connection/NIC

This screams NETWORK problem, and apologies for any unnecessary noise with this post, but I'm at a bit of a loss for a solution. There are no funky network settings inside Windows, nor are there any on the macOS side (other than using Google's 8.8.8.8 and 8.8.4.4 DNS servers), no funky settings in GiA, get_iplayer or the NordVPN app. What am I missing?

Thanks for any advice/assistance!

BBC Download (ID=b0bhrj9b): Downloading Celebrity MasterChef: Series 13 - Episode 4
INFO: 1 total programmes
ERROR: Failed to download URL (3/3): https://open.live.bbc.co.uk/mediaselector/5/select/version/2.0/mediaset/iptv-all/vpid/p06k046y?cb=87541
ERROR: Response: 500 Can't connect to open.live.bbc.co.uk:443
ERROR: Ignore this error if programme download is successful
ERROR: Failed to download URL (3/3): https://open.live.bbc.co.uk/mediaselector/5/select/version/2.0/mediaset/pc/vpid/p06k046y?cb=62054
ERROR: Response: 500 Can't connect to open.live.bbc.co.uk:443
ERROR: Ignore this error if programme download is successful
ERROR: Failed to download URL (3/3): https://open.live.bbc.co.uk/mediaselector/5/select/version/2.0/mediaset/apple-ipad-hls/vpid/p06k046y?cb=64076
ERROR: Response: 500 Can't connect to open.live.bbc.co.uk:443
ERROR: Ignore this error if programme download is successful
ERROR: Failed to download URL (3/3): https://open.live.bbc.co.uk/mediaselector/5/select/version/2.0/mediaset/iptv-all/vpid/b0bhrj7f?cb=63423
ERROR: Response: 500 Can't connect to open.live.bbc.co.uk:443
ERROR: Ignore this error if programme download is successful
ERROR: Failed to download URL (3/3): https://open.live.bbc.co.uk/mediaselector/5/select/version/2.0/mediaset/pc/vpid/b0bhrj7f?cb=94152
ERROR: Response: 500 Can't connect to open.live.bbc.co.uk:443
ERROR: Ignore this error if programme download is successful
ERROR: Failed to download URL (3/3): https://open.live.bbc.co.uk/mediaselector/5/select/version/2.0/mediaset/apple-ipad-hls/vpid/b0bhrj7f?cb=65917
ERROR: Response: 500 Can't connect to open.live.bbc.co.uk:443
ERROR: Ignore this error if programme download is successful
WARNING: No media streams found for requested programme versions and recording modes.
WARNING: The programme may no longer be available - check the iPlayer site.
WARNING: If you are outside the UK, check with the provider to find out if your VPN/VPS/Smart DNS/web proxy has been blocked.
BBC Download (ID=b0bhrj9b): Celebrity MasterChef: Series 13 - Episode 4 Failed

**Comment by alphamikegc** [comment]
Same problem for me, went back to the previous release v.1.13.13 which is allowing download of BBC programmes.

**Comment by dubbayoo** [comment]
That's definitely not my issue, as I've tried a handful of older versions in case the suggestion worked. I know the get_iplayer version inside the Windows 10 VM is an older one, but it doesn't seem to matter which GiA version is in use.

Correction - in double checking the get_iplayer version, it's 3.17.0, as used in GiA 1.13.13.

The issue must be some odd network thing, it's the only explanation I can think of, I just can't think of what the issue would be.

**Comment by dubbayoo** [comment]
Thanks for the reply.

With the VPN enabled on a whitelisted exit point, I confirmed in Safari that iPlayer was watchable before heading to the other URL, and it was. So that's good.

After pasting the URL, I wound up with an XML file as you had predicted. I'll paste the output at the end of this.

For the time being, if this is indeed a GiA issue as it would appear to be (well, not just GiA given get_iplayer 3.17 fails via the CLUI) or a Mac issue perhaps, then I shall continue using 1.13.10 for the time being.

XML file output as follows:

This XML file does not appear to have any style information associated with it. The document tree is shown below.
<mediaSelection xmlns="http://bbc.co.uk/2008/mp/mediaselection">

<media bitrate="1700" encoding="h264" expires="2018-09-30T21:00:00Z" height="396" kind="video" service="stream-uk-iptv_streaming_concrete_combined_sd" type="video/mp4" width="704">
<connection authExpires="2018-09-04T09:17:17+00:00" authExpiresOffset="33117" dpw="34" href="http://vod-hls-uk-live.akamaized.net/usp/auth/vod/piff_abr_full_sd_ad/04853b-b0bhrj7f/vf_b0bhrj7f_1a0fc057-86ee-4f29-9e69-27a720493130.ism.hlsv2.ism/iptv_hd_abr_v1_hls_master.m3u8?__gda__=1536041120_deb507d58f26adcca1fe04af1e9b1477" priority="1" protocol="http" supplier="mf_akamai_uk_hls" transferFormat="hls"/>
<connection authExpires="2018-09-04T05:40:43+00:00" authExpiresOffset="20123" dpw="34" href="http://vod-hls-uk-live.bbcfmt.hs.llnwd.net/usp/auth/vod/piff_abr_full_sd_ad/04853b-b0bhrj7f/vf_b0bhrj7f_1a0fc057-86ee-4f29-9e69-27a720493130.ism.hlsv2.ism/iptv_hd_abr_v1_hls_master.m3u8?s=1535997920&e=1536041120&h=f5aaa3e74dbfc3b8eaaa5946f812fe28" priority="1" protocol="http" supplier="mf_limelight_uk_hls" transferFormat="hls"/>
<connection authExpires="2018-09-04T10:56:22+00:00" authExpiresOffset="39062" dpw="50" href="http://vod-dash-uk-live.akamaized.net/usp/auth/vod/piff_abr_full_sd_ad/04853b-b0bhrj7f/vf_b0bhrj7f_1a0fc057-86ee-4f29-9e69-27a720493130.ism.hlsv2.ism/iptv_hd_abr_v1_dash_master.mpd?__gda__=1536041120_0885621d2479d4ddd76f23baad4371c3" priority="1" protocol="http" supplier="mf_akamai_uk_dash" transferFormat="dash"/>
<connection authExpires="2018-09-04T05:30:09+00:00" authExpiresOffset="19489" href="http://vod-dash-uk-live.bbcfmt.hs.llnwd.net/usp/auth/vod/piff_abr_full_sd_ad/04853b-b0bhrj7f/vf_b0bhrj7f_1a0fc057-86ee-4f29-9e69-27a720493130.ism.hlsv2.ism/iptv_hd_abr_v1_dash_master.mpd?s=1535997920&e=1536041120&h=308f7fa43b5892cf2bbdee1f591e91fc" priority="2" protocol="http" supplier="mf_limelight_uk_dash" transferFormat="dash"/>
<connection authExpires="2018-09-04T09:38:45+00:00" authExpiresOffset="34405" dpw="32" href="http://mm.bidi.bbc.co.uk/vod-hls-uk-live/usp/auth/vod/piff_abr_full_sd_ad/04853b-b0bhrj7f/vf_b0bhrj7f_1a0fc057-86ee-4f29-9e69-27a720493130.ism.hlsv2.ism/iptv_hd_abr_v1_hls_master.m3u8?at=9Ue4QvXxb26713bf1e51204c14d215f98f6f464930486ffb5750069db9000" priority="3" protocol="http" supplier="mf_bidi_uk_hls" transferFormat="hls"/>
<connection authExpires="2018-09-04T11:15:48+00:00" authExpiresOffset="40228" dpw="34" href="https://vod-hls-uk-live.akamaized.net/usp/auth/vod/piff_abr_full_sd_ad/04853b-b0bhrj7f/vf_b0bhrj7f_1a0fc057-86ee-4f29-9e69-27a720493130.ism.hlsv2.ism/iptv_hd_abr_v1_hls_master.m3u8?__gda__=1536041120_deb507d58f26adcca1fe04af1e9b1477" priority="10" protocol="https" supplier="mf_akamai_uk_hls_https" transferFormat="hls"/>
<connection authExpires="2018-09-04T04:36:30+00:00" authExpiresOffset="16270" dpw="34" href="https://vod-hls-uk-live.bbcfmt.hs.llnwd.net/usp/auth/vod/piff_abr_full_sd_ad/04853b-b0bhrj7f/vf_b0bhrj7f_1a0fc057-86ee-4f29-9e69-27a720493130.ism.hlsv2.ism/iptv_hd_abr_v1_hls_master.m3u8?s=1535997920&e=1536041120&h=661d27e5d560a3a5b2b3d9f739978065" priority="10" protocol="https" supplier="mf_limelight_uk_hls_https" transferFormat="hls"/>
<connection authExpires="2018-09-04T11:58:16+00:00" authExpiresOffset="42776" dpw="50" href="https://vod-dash-uk-live.akamaized.net/usp/auth/vod/piff_abr_full_sd_ad/04853b-b0bhrj7f/vf_b0bhrj7f_1a0fc057-86ee-4f29-9e69-27a720493130.ism.hlsv2.ism/iptv_hd_abr_v1_dash_master.mpd?__gda__=1536041120_0885621d2479d4ddd76f23baad4371c3" priority="10" protocol="https" supplier="mf_akamai_uk_dash_https" transferFormat="dash"/>
<connection authExpires="2018-09-04T06:01:05+00:00" authExpiresOffset="21345" href="https://vod-dash-uk-live.bbcfmt.hs.llnwd.net/usp/auth/vod/piff_abr_full_sd_ad/04853b-b0bhrj7f/vf_b0bhrj7f_1a0fc057-86ee-4f29-9e69-27a720493130.ism.hlsv2.ism/iptv_hd_abr_v1_dash_master.mpd?s=1535997920&e=1536041120&h=ec17c0aa3b8718a0b1d0ea991db704af" priority="20" protocol="https" supplier="mf_limelight_uk_dash_https" transferFormat="dash"/>
<connection authExpires="2018-09-04T10:32:07+00:00" authExpiresOffset="37607" dpw="32" href="https://mm.bidi.bbc.co.uk/vod-hls-uk-live/usp/auth/vod/piff_abr_full_sd_ad/04853b-b0bhrj7f/vf_b0bhrj7f_1a0fc057-86ee-4f29-9e69-27a720493130.ism.hlsv2.ism/iptv_hd_abr_v1_hls_master.m3u8?at=r6DKpv4I208ee1f98fd22289898a08ee8b3d67a322c2d3a05750069db9000" priority="30" protocol="https" supplier="mf_bidi_uk_hls_https" transferFormat="hls"/>
</media>
</mediaSelection>

**Comment by Vangelis66** [comment]
@dubbayoo 
Thanks for doing the test I requested and for posting its results 👍 

> With the VPN enabled on a whitelisted exit point, 
> I confirmed in Safari that iPlayer was watchable

.... As has been said before, successfully watching BBC iPlayer in a desktop browser isn't a 100% reliable indication that get_iplayer BBC downloads will also succeed; you may be served MPEG-DASH streams in the browser which work, but GiP prefers HLS streams (and could fail in accessing those...)

> After pasting the URL, I wound up with an XML file as you had predicted.
> (snip) 
>  Suffice it to say the output was as expected.

Well, to me this is sufficient confirmation that your VPN's UK exit node is indeed whitelisted (for GiP purposes) and that you are not suffering a system-wide network mis-configuration, since your browser was successfully served the UK-only version of the mediaselector URL XML response!

The root cause of the issue has to be in GiP's HTTPS protocol implementation on your system, when accessing secure sites, but I don't have a clue currently what this might be... 

> If perl libraries are at fault, mayhaps my still using 
> macOS 10.12.6 could be part of the issue

Quite possible... Browsing the comments of #194, I see very few affected users indicating their OS version, however one (@BruceGoatly) on OS X **10.13**.6 reports unaffected, while another one (@ManolisGledsodakis) on El Capitan (OS X **10.11**.x) reports failure to download; a third user (@ian02054) tried "on a different Mac" (but failed to report OS X version) and succeeded there... 
So yes, the manifestation of the issue appears to be OS version dependent... 

FWIW, https://github.com/get-iplayer/get_iplayer/wiki/osx#before-you-begin states that: 
```
macOS 10.10 (Yosemite) or higher is required
Only macOS system Perl is supported. If you use perlbrew, plenv, ActivePerl, MacPorts Perl, Homebrew Perl, etc., you are responsible for configuring it correctly.
```
In any event, if @skovatch isn't able to pinpoint the culprit, and you are still experiencing the same broken behaviour with the standalone GiP Mac [installer](https://github.com/get-iplayer/get_iplayer_macos/releases/latest), perhaps you should seek help on the GiP support forums, [mac subforum](https://forums.squarepenguin.co.uk/forum-12.html); be sure to first read and abide by their [rules](https://forums.squarepenguin.co.uk/thread-655.html); they check VPN use/IPs/e-mail addresses you register with, and are over all very strict about overseas TV access; please, no mention of UK VPNs and inaccessible TV content; register with your normal IP and say that you can't fetch globally available radio content (which should be also true in your case) ...

**Comment by skovatch** [comment]
> The root cause of the issue has to be in GiP's HTTPS protocol implementation on your system, when accessing secure sites, but I don't have a clue currently what this might be...

Yes, I agree this is definitely the problem. This is why I have been asking folks to install the command-line get_iplayer for comparison. There should be no difference in behavior running that and what's inside GiA.

Also, I hadn't noticed the 10.10 minimum before. I will change GiA to match, since it's currently at 10.9.

**Comment by dubbayoo** [comment]
HTTPS implementation sounds quite likely, since according to https://github.com/tsibley/apple-perl-versions the same version of perl (5.18) has shipped with macOS 10.11, 10.12 and 10.13 (and for that matter, 10.10, which would seem to imply get_iplayer requires 5.18 even if it doesn't explicitly state the version number as per the link above).

Huzzah, a lead! Thanks to all who've contributed, and apologies for not realizing this was a duplicate of #194.

**Comment by MacInWin** [comment]
I am having exactly the same problem of "Can't connect to www.bbc.co.uk:443" and no downloads at all. I'm not tech-savvy and don't want to go through a bunch of hoops to get this working, so is there a layman's fix?

**Comment by ManolisGledsodakis** [comment]
Which version are you using?
My "layman's fix" was to reinstall version 1.13.10, which still works (apparently perfectly). I don't know what's broken in later versions but I'm sticking with this one as long as it works.

**Comment by Vangelis66** [comment]
> Can anyone help me with a link for it ??

https://github.com/Ascoware/get-iplayer-automator/releases/tag/v1.13.10

(the .zip file is what you want...)

**Comment by feverdiva21** [comment]
> > Can anyone help me with a link for it ??
> 
> https://github.com/Ascoware/get-iplayer-automator/releases/tag/v1.13.10
> 
> (the .zip file is what you want...)

Thanks so much - I managed to download 1 programme... now stuck again (maybe this time problem is at vpn) :)

**Comment by stevecopley** [comment]
Have exactly the same issue with v1.13.16: 500 Can't connect to open.live.bbc.co.uk:443
Downgrading to v1.13.10 also fixed things for me
So... Something in more recent versions is making things screwy
(Old iMac running OS X 10.11.6)

**Comment by zugernaut** [comment]
Apologies. I posted some results of my own investigation into the difference between 1.13.10 and 1.13.16 under issue #211.

**Comment by zugernaut** [comment]
Apologies. I posted some results of my own investigation into the difference between 1.13.10 and 1.13.16 under issue #211.

**Comment by dubbayoo** [comment]
> 'Updated the root certificates used by Mozilla::CA. This should address the 500 errors when starting a BBC download' (release notes of 1.13.17)
> Tried this new release, but still no go for me with the same 500…. 443 error as before. Fixed by rolling back to 1.13.10

Ditto. Tried 1.13.17 and the errors were still as rampant as ever. Reverting to 1.13.10 works like a charm as ever.

**Comment by stevecopley** [comment]
That doesn't make sense... All downloads work with 1.13.10. No downloads work with later versions (500 errors). So how is it the VPN service at fault if the older version works just fine with the VPN?

**Comment by ManolisGledsodakis** [comment]
I concur with Steve. exactly the result I get. 1.13.10 works. Later versions don't (with the same VPN). 

Something else is going on here. When I get time, I'll set up my third computer with the latest GiA and see what that does but I don't expect anything different. As Einstein allegedly said: doing the same thing over and expecting different results is a sign of insanity!

**Comment by dubbayoo** [comment]
I initially opened this thread because of inconsistencies in operation; I couldn’t use GiA to download on the Mac side while get_iplayer worked fine inside a Windows VM. Rolling back to 1.13.10 was the fix then, and remains to be the fix for me and for others. If 1.13.10 used a buggy PERL implementation then its flaw is its saving grace, at least for some (several) people, and the issue is not VPN-related. Closure of this thread for VPN reasons isn’t proper. If it’s an issue with macOS or PERL or a mixture of the two then so be it, close it for that reason if it can’t be fixed within GiA (which, seemingly it can, albeit in a potentially undesirable manner). 

I would be curious, however, to learn of why some people don’t have this problem, and why their configuration allows success with iterations of GiA beyond 1.13.10, but of course those who don’t have problems don’t often tend to look at threads or resources like this one. :)

**Comment by ian02054** [comment]
I agree, and was just abut to post a similar message. 1.13.10 works fine, and nothing since has done for me. I think it must be more than a vpn issue if so many people have it working successfully with that release, but nothing beyond.

**Comment by mastige** [comment]
Thanks for the reply. I understand that you believe that BBC is blocking the downloads. With exactly the same settings I can watch shows in iPlayer. Is this consistent with BBC blocking? I do appreciate all your efforts, but remain puzzled by this issue.
